### PR TITLE
Deprecate Firecracker v1.2

### PR DESCRIPTION
**Pull request by pb8o**
## Changes

Deprecate v1.2 by raising the minimum supported version to 1.3

## Reason

Per release policy after v1.4 was released

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/pytest.ini [hunk 1]
```diff
@@ -5,6 +5,7 @@ addopts =
     --durations=10
     --showlocals
     -m 'not nonci and not no_block_pr'
+    --json-report --json-report-file=../test_results/test-report.json
```
Comment (bchalios) [inline tests/pytest.ini:8]: nit: Could you add a bit more context on the commit message, on what is this option, what effect does it have etc.?

Comment (pb8o) [reply tests/pytest.ini:8]: Fixed, but let me know if you have questions. It may be easier to just show you an example report.

**Review by roypat — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]

**Comment by sudanl0** [comment]
should this be backported to the supported release branches as well? like 1.4/1.3?

**Comment by pb8o** [comment]
> should this be backported to the supported release branches as well? like 1.4/1.3?

To 1.4 yes, to 1.3 I'm not sure, since we probably still want to test 1.2 -> 1.3

**Pull request merged**